listening-comp/backend/speech_recognition.py

```
@@ -0,0 +1,174 @@
+        if not segments:
+            return 0.0
+            
+        confidences = [segment.get("confidence", 0) for segment in segments]
+        return sum(confidences) / len(confidences)
+    
+    def get_model_info(self) -> Dict[str, Any]:
+        """Get information about the current model."""
+        return {
+            "model_size": self.model_size,
+            "is_local": self.use_local_model,
+            "device": "cuda" if torch.cuda.is_available() else "cpu",
```

Ruff (0.8.2)

Undefined name `torch`

(F821)